### DISCUSSION TITLE
Is it possible to wait for revalidatePath to complete?

### DISCUSSION THREAD

**Discussion by asnaeb** [vercel/next.js#53245] [Help]
### Summary

In AppRouter, when calling `revalidatePath` in a server action, is there any way to know when the revalidation has been completed? Let's say we have a server component fetching data and passing it down to a client component in which we are calling a server action which calls `revalidatePath` on that path. Is there any way to know when the revalidation has been completed and the data has been refetched?

**Comment by unav4ila8le** [comment]
I would like to know this as well. I have been trying to find a solution for a while now, but it seems like there is no official Next.js API to track revalidatePath completion. Is there any officially supported solution for this? What would be the ideal practice for when we have a form submission which invokes a server action with a revalidation after success? Cause I do not really like the idea of having a stale UI for few seconds before the data is refreshed...

Did you find any solution @asnaeb ?
